File: frontend/src/types/fine.ts
export type FineStatus = "Unpaid" | "Paid" | "Waived";
export type FineReason = "Overdue" | "Lost" | "Damaged";

export interface Fine {
  id: number;
  borrowRecordId: number;
  bookTitle: string;
  userId: number;
  userName: string;
  amount: number;
  reason: FineReason;
  status: FineStatus;
  createdAt: string;
  paidAt?: string | null;
}

File: backend/Domain/Entities/Fine.cs
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities;

public class Fine : BaseEntity
{
    public decimal Amount { get; set; }
    public FineReason Reason { get; set; } = FineReason.Overdue;
    public FineStatus Status { get; set; } = FineStatus.Unpaid;
    public DateTime? PaidAt { get; set; }

    public int BorrowRecordId { get; set; }
    public BorrowRecord? BorrowRecord { get; set; }
}
